new documentation on type inferring with fq (#13807)
ydb-platform/ydb: ydb/docs/en/core/concepts/federated_query/s3/external_data_source.md

Year Int32 NOT NULL
```

> If a data field is marked as required (`NOT NULL`) but this field is missing in the processed file, the processing of such a file will result in an error. If a field is marked as optional (`NULL`), no error will occur in the absence of the field in the processed file, but the field will take the value `NULL`. Keyword `NULL` is optional in this context.

Reviewer: ```suggestion
If a data field is marked as required (`NOT NULL`) but is missing in the processed file, processing such a file will result in an error. If a field is marked as optional (`NULL`), no error will occur if the field is absent in the processed file, but the field will take the value `NULL`. The keyword `NULL` is optional in this context.
```

### Schema inference {#inference}

> 

Reviewer 2: Необходимо коротко описать, что такое schema inference.

Reviewer 2: {{ ydb-short-name }} can determine the data schema of the files inside the bucket so that you do not have to specify these fields manually.

> ### Schema inference {#inferring}

Reviewer: ```suggestion
### Schema inference {#inference}
```

> Schema inference is available for all [data formats](formats.md#formats) except `raw` and `json_as_string`. It can be helpful when the schema contains a large number of fields. To avoid entering those fields manually, use the `WITH_INFER` parameter.:

Reviewer 2: ```suggestion

{% note info %}

Schema inference is available for all [data formats](formats.md#formats) except `raw` and `json_as_string`. For these formats you must [describe the schema manually](#schema). 

{% endnote %}

To enable schema inference, use the `WITH_INFER` parameter:
```

> Schema inference is available for all [data formats](formats.md#formats) except `raw` and `json_as_string`. It can be useful when the schema contains a large number of fields. In order not to enter those fields manually, use the `WITH_INFER` parameter.:

Reviewer: ```suggestion
Schema inference is available for all [data formats](formats.md#formats) except `raw` and `json_as_string`. It can be helpful when the schema contains a large number of fields. To avoid entering those fields manually, use the `WITH_INFER` parameter:
```

```yql
SELECT
  <expression>
FROM
  <object_storage_connection_name>.`<file_path>`
WITH(
  FORMAT = "<file_format>",
  COMPRESSION = "<compression>",
  WITH_INFER = "true")
WHERE
  <filter>;
```

Where:

* `object_storage_connection_name` — the name of the external data source leading to the S3 bucket ({{ objstorage-full-name }}).

> * `file_path` — the path to the file or files inside the bucket. Wildcards `*` are supported; more details are available [below](#path_format).

Reviewer 2: ```suggestion
* `file_path` — the path to the file or files inside the bucket. Wildcards `*` are supported. For more information, see [{#T}](#path_format).
```

> Where:
> 
> * `object_storage_connection_name` — the name of the external data source leading to the S3 bucket ({{ objstorage-full-name }}).
> * `file_path` — the path to the file or files inside the bucket. Wildcards `*` are supported; more details [in the section](#path_format).
> * `file_format` — the [data format](formats.md#formats) in the files. All formats except for `raw` and `json_as_string` are supported.
> * `compression` — the [compression format](formats.md#compression_formats) of the files.
> 
> As a result of executing such a query, the names and types of fields will be inferred.

Reviewer: ```suggestion
Where:

* `object_storage_connection_name` — the name of the external data source leading to the S3 bucket ({{ objstorage-full-name }}).
* `file_path` — the path to the file or files inside the bucket. Wildcards `*` are supported; more details are available [below](#path_format).
* `file_format` — the [data format](formats.md#formats) in the files. All formats except `raw` and `json_as_string` are supported.
* `compression` — the [compression format](formats.md#compression_formats) of the files.

As a result of executing such a query, the names and types of fields will be inferred.
```
